src/Ampersand/Core/Relation.php

```diff
@@ -339,4 +356,33 @@ public function empty(): void
         }
         $this->logger->info("Deleted all links in relation: {$this}");
     }
+
+    public function hasDefaultSrcValues(): bool
+    {
+        return !empty($this->defaultSrc);
+    }
+
+    public function hasDefaultTgtValues(): bool
+    {
+        return !empty($this->defaultTgt);
+    }
+
+    public function getDefaultSrcValues(): array
+    {
+        return array_map('self::resolveDefaultValue', $this->defaultSrc);
+    }
+
+    public function getDefaultTgtValues(): array
+    {
+        return array_map('self::resolveDefaultValue', $this->defaultTgt);
+    }
+
+    protected static function resolveDefaultValue(string $value): string
+    {
+        if (substr($value, 0, 5) === '{php}') {
+            $code = 'return('.substr($value, 5).');';
+            $value = (string) eval($code);
+        }
+        return $value;
+    }
```

Comment:
This suggests that the php part will return exactly one atom. Is this intended? What if the function returns multiple atoms? Should this behaviour be documented for the modeller?

Reply:
You're right. We discussed that the eval may return multiple values. I'll update the code.

Reply:
To reduce complexity in the code I propose to allow only a single scalar value to be returned by the evalphp function.  I've added a check for this with proper error message if this goes wrong. If multiple values are needed, this can be done by specifying multiple `EVALPHP` parts (see below). I think this is an edge case anyway.

```adl
RELATION projectStatus[Project*ProjectStatus] [UNI,TOT] DEFAULT TGT EVALPHP "test()" TGT EVALPHP "test2()"
```